homeassistant/components/imeon_inverter/sensor.py

```diff
@@ -0,0 +1,389 @@
+"""Imeon inverter sensor support."""
+
+import logging
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import InverterCoordinator
+
+type InverterConfigEntry = ConfigEntry[InverterCoordinator]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+ENTITY_DESCRIPTIONS = (
+    # Battery
+    SensorEntityDescription(
+        key="battery_autonomy",
+        name="Battery Autonomy",
+        native_unit_of_measurement="",
+        icon="mdi:battery-clock",
+    ),
+    SensorEntityDescription(
+        key="battery_charge_time",
+        name="Battery Charge Time",
+        native_unit_of_measurement="",
+        icon="mdi:battery-charging",
+    ),
+    SensorEntityDescription(
+        key="battery_power",
+        name="Battery Power",
+        native_unit_of_measurement="W",
+        icon="mdi:battery",
+    ),
+    SensorEntityDescription(
+        key="battery_soc",
+        name="Battery SOC",
```

Comment:
also check https://developers.home-assistant.io/docs/core/entity/sensor/#available-device-classes and apply where appropriate